marshaling.go

```diff
@@ -219,9 +209,23 @@ type convertCtx struct {
 	sourceType      Type
 	destFieldName   string
 	destType        reflect.Type
+	scanner         *structScanner
```

Comment:
We shouldn't add `scanner` here, because `convertCtx` is a shorter-lived object than the `scanner`, and hence we're creating a parent -> child relationship which inverses the normal lifecycle, which is error prone.

Instead, let's change the signature of `structScanConvert(ctx convertCtx) (reflect.Value, error)` to `structScanConvert(dests map[string]interface{}, ctx convertCtx) (reflect.Value, error)`

Reply:
makes sense.

passing in the ctx definitely not best, but it also allows deep calling of structScan(). will rework that so that most of the structScan code now actually lives in the structScanConvert for *Worksheet